### PR TITLE
[PAID]nanite logic

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
nanite programs now have a logic field, which can be set to AND, NAND, OR or NOR. This changes the way nanites interpret their rules, in a way proper for each of the logic types. currently, nanites only have AND logic.

## Why It's Good For The Game
allows more complexity for nanite programs

## Changelog
:cl:
add: added OR, NOR, and NAND logic types to nanite programs
/:cl:

### CODE REVIEW

**Comment by Kapu1178** [comment]
Originally nanites operated exclusively on AND logic. I tested these changes extensively and they all work as intended.

Very based and i will enjoy making even more complicated nanites programs

**Comment by Sinestia** [comment]
Nanites are already extremely strong and I'm not 100% sure if even more power is warranted/good

 But maybe it helps circuitbois cope if that mechanic is lost by adding more control flow stuff to the game so its not the worst thing

**Comment by Kapu1178** [comment]
> 
> 
> Nanites are already extremely strong and I'm not 100% sure if even more power is warranted/good
> 
> But maybe it helps circuitbois cope if that mechanic is lost by adding more control flow stuff to the game so its not the worst thing

Fat disagree here. Nanites are very easy to counter both ways, and do not nearly reach the height of other powergaming divisions like Chem, Toxins, or even Mining.

File: code/modules/research/nanites/nanite_programs.dm [hunk 1]
```diff
@@ -54,6 +54,15 @@
 	//Rules that automatically manage if the program's active without requiring separate sensor programs
 	var/list/datum/nanite_rule/rules = list()
 
+	//Logic
+	//a list of logic types a nanite program's rules follow
+	var/list/static/logic = list(
+    	"AND" = NL_AND,
```
Comment (Archanial) [inline code/modules/research/nanites/nanite_programs.dm]: "AND" = "AND"?

**Comment by Sinestia** [comment]
> > Nanites are already extremely strong and I'm not 100% sure if even more power is warranted/good
> > But maybe it helps circuitbois cope if that mechanic is lost by adding more control flow stuff to the game so its not the worst thing
> 
> Fat disagree here. Nanites are very easy to counter both ways, and do not nearly reach the height of other powergaming divisions like Chem, Toxins, or even Mining.

>Chem
Ruko's slapping it
>Toxins
Good for single traitors with hijack, otherwise unusable
>Mining
Was hit, and seems like it's going to be hit again

 It's more the idea of having strong passive healing across the station, especially if species stuff is fixed, displacing medbay. Especially since we have none of the nerfs, still have adrenals, gamer neural reimaging, dermal armor in its old glory, etc.. 

 Its not even powergaming, that's a meme, it's about how distributable, dependent on one person for production at one stage, etc. those benefits are.  

 Stuff that's more counterable has been nerfed in the past like secborgs. 

Not that im saying nerf or even reject this im just saying I'm MEH on it 


Also hella bias >;) >;) >;) >;) >;) >;)

File: code/modules/research/nanites/nanite_programs.dm (hunk 1 above)
Comment (zeskorion) [reply code/modules/research/nanites/nanite_programs.dm]: yes. something something magic strings bad

Comment (Archanial) [reply code/modules/research/nanites/nanite_programs.dm]:
```suggestion
    	NL_AND,
```

**Comment by r1ks-iwnl** [comment]
>Nanites are very easy to counter

Nanites main weakness is that they can blow you up at any time which WON'T happen unless you have murderbone objectives or else you'll get banned.

**Comment by aidan-bowman** [comment]
> > Nanites are very easy to counter
> 
> Nanites main weakness is that they can blow you up at any time which WON'T happen unless you have murderbone objectives or else you'll get banned.

wont happen? bruh what kinda planet are you livin on lol

**Review by utilbird — APPROVED** [review]

**Pull request merged**